cibuildwheel/windows.py

```diff
@@ -136,12 +131,27 @@ def shell(args, env=None, cwd=None):
 
         # run the before_build command
         if before_build:
-            before_build_prepared = prepare_command(before_build, project=abs_project_dir)
+            before_build_prepared = prepare_command(before_build, {'project': abs_project_dir})
             shell([before_build_prepared], env=env)
 
         # build the wheel
+        if os.path.exists(built_wheel_dir):
+            shutil.rmtree(built_wheel_dir)
+        os.makedirs(built_wheel_dir)
         shell(['pip', 'wheel', abs_project_dir, '-w', built_wheel_dir, '--no-deps'] + get_build_verbosity_extra_flags(build_verbosity), env=env)
-        built_wheel = glob(built_wheel_dir+'/*.whl')[0]
+        built_wheel = glob(os.path.join(built_wheel_dir, '*.whl'))[0]
+
+        # delocate the wheel
+        if os.path.exists(delocated_wheel_dir):
+            shutil.rmtree(delocated_wheel_dir)
+        os.makedirs(delocated_wheel_dir)
+        if built_wheel.endswith('none-any.whl') or not delocate_command:
+            # pure Python wheel or empty delocate command
+            shutil.move(built_wheel, delocated_wheel_dir)
+        else:
+            delocate_command_prepared = prepare_command(delocate_command, {'wheel': built_wheel, 'dest_dir': delocated_wheel_dir})
+            shell([delocate_command_prepared], env=env)
+        delocated_wheel = glob(os.path.join(delocated_wheel_dir, '*.whl'))[0]
```

Comment:
Agreeing with @joerick here; the name "delocate" makes absolutely no sense in a Windows (or Linux) context.

Reply:
Replaced with `repair`/`repaired` everywhere.

Reply:
Still not 100% convinced that we don't need `wheel` in those names, but I'll leave that decision up to @joerick.